### PR TITLE
Add A/B test mode to local devicelab runner

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

Add a new option for running devicelab tasks in an A/B mode for verifying engine performance optimizations.

### CODE REVIEW

File: dev/devicelab/README.md [hunk 1]
```diff
@@ -162,6 +162,47 @@ flags to `bin/run.dart`:
...
+Score	Average A (noise)	Average B (noise)	Speed-up
+bench_card_infinite_scroll.canvaskit.drawFrameDuration.average	2900.20 (8.44%)	2426.70 (8.94%)	1.20x
+bench_card_infinite_scroll.canvaskit.totalUiFrame.average	4964.00 (6.29%)	4098.00 (8.03%)	1.21x
+draw_rect.canvaskit.windowRenderDuration.average	1959.45 (16.56%)	2286.65 (0.61%)	0.86x
+draw_rect.canvaskit.sceneBuildDuration.average	1969.45 (16.37%)	2294.90 (0.58%)	0.86x
+draw_rect.canvaskit.drawFrameDuration.average	5335.20 (17.59%)	6437.60 (0.59%)	0.83x
+draw_rect.canvaskit.totalUiFrame.average	6832.00 (13.16%)	7932.00 (0.34%)	0.86x
+```
+
+The outpout contains averages and noises for each score. More importantly, it
```
Comment (christopherfujino) [inline dev/devicelab/README.md:200]: sp. `outpout -> output`

File: dev/devicelab/bin/run.dart [hunk 2]
```diff
@@ -9,18 +9,34 @@ import 'dart:io';
...
+
+/// The build of the local engine to use.
+///
+/// Required for A/B test mode.
+String localEngine;
+
+/// The path to the engine "src/" directory.
+String localEngineSrcPath;
+
+/// Whether to exist on first test failure.
```
Comment (christopherfujino) [inline dev/devicelab/bin/run.dart:32]: sp `exist -> exit`

File: dev/devicelab/README.md (hunk 1 above)
Comment (yjbanov) [reply dev/devicelab/README.md]: Done.

File: dev/devicelab/bin/run.dart (hunk 2 above)
Comment (yjbanov) [reply dev/devicelab/bin/run.dart]: Done.

**Review by christopherfujino — APPROVED** [review]
I didn't test this locally, but the code LGTM. Neat feature!

File: dev/devicelab/bin/run.dart [hunk 3]
```diff
@@ -9,18 +9,34 @@ import 'dart:io';
...
 import 'package:flutter_devicelab/framework/manifest.dart';
 import 'package:flutter_devicelab/framework/runner.dart';
 import 'package:flutter_devicelab/framework/utils.dart';
 
+ArgResults args;
+
 List<String> _taskNames = <String>[];
 
+/// Suppresses standard output, prints only standard error output.
+bool silent;
```
Comment (ferhatb) [inline dev/devicelab/bin/run.dart:22]: suggestion:  verbose with default to true so no-verbose would make it silent.

**Pull request merged**